lsp-mode.el

```diff
@@ -3355,17 +3364,37 @@ disappearing, unset all the variables related to it."
                            (ht-keys created-watches))))
       ;; create watch for each root folder without such
       (dolist (folder root-folders)
-        (let ((watch (make-lsp-watch :root-directory folder)))
+        (let* ((watch (make-lsp-watch :root-directory folder))
+               (ignored-things (lsp--get-ignored-regexes-for-workspace-root folder))
+               (ignored-files-regex-list (car ignored-things))
+               (ignored-directories-regex-list (cadr ignored-things))
+               )
           (puthash folder watch created-watches)
           (lsp-watch-root-folder (file-truename folder)
                                  (-partial #'lsp--file-process-event (lsp-session) folder)
+                                 ignored-files-regex-list
+                                 ignored-directories-regex-list
                                  watch
                                  t)))))
 
   (push
    (make-lsp--registered-capability :id id :method method :options register-options?)
    (lsp--workspace-registered-server-capabilities lsp--cur-workspace)))
 
+(defun lsp--get-ignored-regexes-for-workspace-root (workspace-root)
+  "Return a list of the form (lsp-file-watch-ignored-files lsp-file-watch-ignored-directories) for the given WORKSPACE-ROOT."
+  ;; The intent of this function is to provide per-root workspace-level customization of the
+  ;; lsp-file-watch-ignored-directories and lsp-file-watch-ignored-files variables.
+  (let* (
```

Comment:
Can you use the snippet from https://github.com/emacs-lsp/lsp-mode/pull/2736#issuecomment-803670625 ?

The advantage is that it does not run any irrelevant hooks like fundamental-mode-hook, find-file-hooks and also it does not access the disk.

Reply:
switched to `with-temp-buffer` as per that comment -- tested that it worked even if dir locals are set via the class.